src/types.rs

```diff
@@ -77,30 +54,31 @@ impl LineType {
 
 #[cfg(test)]
 mod test {
-    extern crate rustc_serialize;
-    extern crate docopt;
+    extern crate serde_json;
 
     use super::OsType::{self, Linux, OsX, SunOs, Other};
     use super::LineType;
-    use rustc_serialize::json;
+    use self::serde_json::from_str;
 
     #[test]
     fn test_os_type_decoding_regular() {
-        assert_eq!(json::decode::<OsType>("\"linux\"").unwrap(), Linux);
-        assert_eq!(json::decode::<OsType>("\"osx\"").unwrap(), OsX);
-        assert_eq!(json::decode::<OsType>("\"sunos\"").unwrap(), SunOs);
-        assert_eq!(json::decode::<OsType>("\"other\"").unwrap(), Other);
+        assert_eq!(from_str::<OsType>("\"linux\"").unwrap(), Linux);
+        assert_eq!(from_str::<OsType>("\"osx\"").unwrap(), OsX);
+        assert_eq!(from_str::<OsType>("\"sunos\"").unwrap(), SunOs);
+        assert_eq!(from_str::<OsType>("\"other\"").unwrap(), Other);
     }
 
-    #[test]
-    fn test_os_type_decoding_uppercase() {
-        assert_eq!(json::decode::<OsType>("\"Linux\"").unwrap(), Linux);
-        assert_eq!(json::decode::<OsType>("\"LINUX\"").unwrap(), Linux);
-    }
+    // REVIEW: Not sure how to do case-insensitive deserialization with serde
+    //         json. Ok to ditch support for this?
+    //#[test]
+    //fn test_os_type_decoding_uppercase() {
+    //    assert_eq!(from_str::<OsType>("\"Linux\"").unwrap(), Linux);
+    //    assert_eq!(from_str::<OsType>("\"LINUX\"").unwrap(), Linux);
+    //}
```

Comment:
I think we're looking for this, right? https://github.com/serde-rs/serde/issues/586

I'm not sure how important it is to have case insensitive matching, but the workaround by dtolnay seems quite low-effort to implement, what do you think?

Reply:
Looks simple enough, but would also be nice to avoid the extra code if it's not needed. I can add the workaround if you would like to keep the support for case-insensitive input?